beaconApp/src/main/java/com/pulselink/beacon/data/SmsSyncManager.kt

```
@@ -0,0 +1,224 @@
+
+            // Sync threads
+            threads.forEach { thread ->
+                syncThread(uid, thread)
+                // Sync messages for each thread
+                val messages = repository.messagesForThread(thread.threadId, limit = 50)
+                messages.forEach { msg ->
+                    syncMessage(uid, msg)
+                }
+            }
+            Log.d("SmsSyncManager", "Full sync complete")
+        } catch (e: Exception) {
```

detekt (1.23.8)

[warning] The caught exception is too generic. Prefer catching specific exceptions to the case that is currently handled.

(detekt.exceptions.TooGenericExceptionCaught)